### PR TITLE
doc: use recommended shebang approach in documentation code block

### PR DESCRIPTION
**Pull request by hackerrdave**
Documentation update to use recommended shebang approach mentioned in the [developer notes](https://github.com/bitcoin/bitcoin/blob/master/doc/developer-notes.md#shebang)

### CODE REVIEW

**Review by promag — COMMENTED** [review]
Have you searched for similar places were this change could be applied?

**Comment by hackerrdave** [comment]
@promag Yea, the only other place I believe that this change could be applied is: https://github.com/bitcoin/bitcoin/blob/master/test/lint/lint-shell.sh#L54

I'll add that change as well

**Comment by fanquake** [comment]
Please [squash your commits](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#squashing-commits).

File: test/lint/lint-shell.sh [hunk 1]
```diff
@@ -51,7 +51,7 @@ do
     echo
     echo "$descriptor"
     # Use #!/bin/bash as gitian-builder/bin/gbuild does to complete a script.
```
Comment (laanwj) [inline test/lint/lint-shell.sh]: pleas update the comment as well

Comment (hackerrdave) [reply test/lint/lint-shell.sh]: updated comment

File: test/lint/lint-shell.sh [hunk 2]
```diff
@@ -50,8 +50,8 @@ for descriptor in $(git ls-files -- 'contrib/gitian-descriptors/*.yml')
 do
     echo
     echo "$descriptor"
-    # Use #!/bin/bash as gitian-builder/bin/gbuild does to complete a script.
-    SCRIPT=$'#!/bin/bash\n'$(yq -r .script "$descriptor")
+    # Use #!/usr/bin/env bash to complete script with preferred shebang approach.
+    SCRIPT=$'#!/usr/bin/env bash\n'$(yq -r .script "$descriptor")
```
Comment (hebasto) [inline test/lint/lint-shell.sh]: Please do not change these lines as it is intended to mimic the way `gitian-builder/bin/gbuild` does: https://github.com/bitcoin/bitcoin/pull/17361#discussion_r342189958

**Review by hebasto — CHANGES_REQUESTED** [review]

File: test/lint/lint-shell.sh (hunk 1 above)
Comment (hebasto) [reply test/lint/lint-shell.sh]: https://github.com/bitcoin/bitcoin/pull/17751#discussion_r358908378

File: test/lint/lint-shell.sh (hunk 2 above)
Comment (hackerrdave) [reply test/lint/lint-shell.sh]: ah gotcha! Thank you for the clarification - will remove this change

**Review by hebasto — APPROVED** [review]
ACK 6094222de7820d235e6e8c66e589aa71db08c077, I have reviewed the code, and it looks OK, I agree it can be merged.

**Pull request merged**